lighthouse-core/computed/main-thread-tasks.js

```diff
@@ -177,27 +173,35 @@ class MainThreadTasks {
        */
       case 'v8.compile':
       case 'EvaluateScript':
+        if (task.event.args.data) {
+          taskURLs = [task.event.args.data.url].concat(
+            (task.event.args.data.stackTrace || []).map(entry => entry.url));
```

Comment:
IMO, moving these prefixes and `||` conditions down into the switch makes it harder to figure out what's going on for each event type. Since we have the generic catchall in the `default` case anyway, would a `'stackTrace' in data` at the top be able to do the same type checking job here?

Reply:
if you do that before the type narrowing done in the switch statement, you lose out on the type checking aspect. for example, you can accidentally try to use stackFrames in a trace type that does not have it (although it would not error at runtime)

![image](https://user-images.githubusercontent.com/4071474/55205877-db212100-5191-11e9-9e04-9bb53c0f5072.png)

^ FunctionCall doesn't have stackTrace data.

thoughts on this?

![image](https://user-images.githubusercontent.com/4071474/55206138-a6fa3000-5192-11e9-8f80-98ce81988a55.png)